ledger/acctupdates_test.go

```diff
@@ -1352,6 +1352,191 @@ func TestBoxNamesByAppIDs(t *testing.T) {
 	}
 }
 
+func TestBoxCache(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	initialBlocksCount := 1
+	accts := make(map[basics.Address]basics.AccountData)
+
+	protoParams := config.Consensus[protocol.ConsensusCurrentVersion]
+	protoParams.MaxBalLookback = 8
+	protoParams.SeedLookback = 1
+	protoParams.SeedRefreshInterval = 1
+	testProtocolVersion := protocol.ConsensusVersion("test-protocol-TestBoxNamesByAppIDs")
+	config.Consensus[testProtocolVersion] = protoParams
+	defer func() {
+		delete(config.Consensus, testProtocolVersion)
+	}()
+
+	ml := makeMockLedgerForTracker(t, true, initialBlocksCount, testProtocolVersion,
+		[]map[basics.Address]basics.AccountData{accts},
+	)
+	defer ml.Close()
+
+	conf := config.GetDefaultLocal()
+	au, _ := newAcctUpdates(t, ml, conf, ".")
```

Comment:
@algoidurovic As a heads up, I created a tangentially related PR:  https://github.com/algorand/go-algorand/pull/4353.  If https://github.com/algorand/go-algorand/pull/4353 is merged ahead of this PR, then a minor change will be needed.

Reply:
@algoidurovic FYI - #4353 is merged.  When you merge master in, you'll have minor compile-time errors to resolve.  Let me know if you'd like to pair.

Reply:
I think this can be resolved now.